src/test/ui/issues/issue-51714.stderr

```diff
@@ -22,14 +22,6 @@ error[E0572]: return statement outside of function body
 LL |     [(); return while let Some(n) = Some(0) {}];
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-warning: irrefutable while-let pattern
```

Comment:
Now that I think of it.... This is actually a diagnostics improvement. This now-removed diagnostic is just plain wrong.